### ISSUE TITLE
Implement CI/CD pipeline to verify different Kyma upgrade-paths using Kyma CLI(1.x -> main, 2.x -> main etc.)

### ISSUE THREAD

**Issue by tobiscr** [kyma-project/cli#1117]
**Description**

The current assumption is that the latest CLI (including an up-to-date reconciler library) is capable to deal with different upgrading scenarios.

A CI/CD pipeline is required which verifies that this upgrade paths are verified and properly working. It has to possible to verify multiple migration paths in parallel to reduce processing time: this can be achieved by either by creating multiple CI/CD pipelines or by starting multiple upgrade procedures on one machine but using different target K8s clusters (e.g. using multiple K3d instances on one machine).

Following upgrade paths have to be covered at the very beginning:

* 1.24.8 -> main
* 2.0.0-rcX (latest official 2.0 release) -> main

AC:
* Following constraints are configurable for each upgrade path:
    * Which Kyma CLI version is used for creating and for upgrading the cluster (e.g. previous cluster is created with CLI version 1.24.8 and upgrade has to happen with latest Kyma CLI)
    * The executed CLI command (required as the 1.X CLI requires different parameters as the 2.X CLI)
* The logs of the installation and upgrade process have to be easily retrievable and are separately logged (no scrumbling of log-message if we run multiple upgrade-processes in parallel)

**Reasons**

Ensure working upgrade paths when using Kyma CLI.

**Attachments**

**Comment by tobiscr** [comment]
Duplicate of https://github.com/kyma-project/kyma/issues/12819
